main.tf

```diff
@@ -23,6 +23,11 @@ locals {
 
   # set key_protect_guid as either the ID of the passed in name of instance or the one created by this module
   kms_guid = var.create_key_protect_instance ? module.key_protect[0].key_protect_guid : var.existing_kms_instance_guid
+
+  # set key_protect_crn as either the crn of the passed in name of instance or the one created by this module
+  kms_crn = var.create_key_protect_instance ? module.key_protect[0].key_protect_crn : var.existing_kms_instance_crn
+  # tflint-ignore: terraform_unused_declarations
+  cbr_validation = length(regexall(".*hs-crypto.*", local.kms_crn)) > 0 && length(var.cbr_rules) > 0 ? tobool("When passing an HPCS instance as value for `existing_kms_instance_crn` you cannot provide `cbr_rules`. Context-based restrictions is not supported for HPCS instances. For more information, see https://cloud.ibm.com/docs/account?topic=account-context-restrictions-whatis#cbr-adopters for the services that supports CBR") : true
```

Comment:
wait hang on @dishankkalra23 , in https://github.ibm.com/GoldenEye/issues/issues/9432#issuecomment-84352596 your saying HPCS does support CBRs. So if this is the case, we should allow the module to create CBR rules in an existing HPCS instance

Reply:
Let's confirm this with the HPCS team because HPCS was not listed in CBR-supported services in the documentation: https://cloud.ibm.com/docs/account?topic=account-context-restrictions-whatis#cbr-adopters

Reply:
According to this thread: https://ibm-cloudplatform.slack.com/archives/CFFC7M3B3/p1720764415768319
CBR restriction on HPCS is not available.